lib/stripe/api.ex

```diff
@@ -217,6 +217,22 @@ defmodule Stripe.API do
     end
   end
 
+  @spec add_telemetry_metadata(list, map) :: list
+  defp add_telemetry_metadata(opts, args) do
+    parsed_uri = URI.parse(args.req_url)
+    api_version = args.req_headers["Stripe-Version"]
+
+    opts
+    |> Keyword.put(:telemetry_metadata, %{
+      stripe_api_endpoint: parsed_uri.path,
```

Comment:
This could be extracted from `http_url` if somebody really wants to map the data to something else